### ISSUE TITLE
readOne operation in v4 does not have id argument

### ISSUE THREAD

**Issue by bswan** [silverstripe/silverstripe-graphql#311]
In v3  readOne operation had id argument so it was easy to use to get exact instance of object you require.

Like that:

```
readOneBook(id:4){
 title
 author
 summary
}
```

Now, without dedicated field, you need to use filter argument insterad:

```
readOneBook(filter:{id:{eq:4}}) {
  title
  author
  summary
}
```

It is too wordy for such simple task

**Comment by chillu** [comment]
Thank you for the feedback! Unfortunately that's one of the tradeoffs from having an ORM that allows the creation of complex filters. You can add [custom queries](https://docs.silverstripe.org/en/4/developer_guides/graphql/working_with_generic_types/building_a_custom_query/) with your own arguments. 

I'd rather work on a smoother transition path from model types with built-in plugins like versioned, filtering, permissions, to custom queries. But that's not really part of the initial release.

So, while I agree with you that stylistically it's not as nice, it's also a much more powerful way of querying. Given everything else that needs to be solved in this module, I don't see the very few active contributors getting around to this enhancement anytime soon. You're welcome to dig into the plugin structure and have a think about how this could apply to custom queries.
